### PR TITLE
CORE-5744 Changes required after membership API clean up

### PR DESCRIPTION
**Pull request by charlieR3**
Adjust `runtime-os` to handle `DEFAULT_PROTOCOL_VERSION` for endpoints moved off of the public interface for `EndpointInfo`. The `impl` sets it's own defaults now instead.

API PR: https://github.com/corda/corda-api/pull/553

### CODE REVIEW

**Review by nikinagy — DISMISSED** [review]
LGTM

File: libs/membership/membership-impl/src/test/kotlin/net/corda/membership/lib/impl/EndpointInfoImplTest.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        private const val URL = "https://localhost:10000"
+        private const val HOST = "localhost"
+        private const val PORT = 10000
+        private const val DEFAULT_PROTOCOL_VERSION = 1
+    }
+
+    @Test
+    fun `creating EndpointInfo using default protocol version`() {
+        val endpointInfo = EndpointInfoImpl(URL)
+        assertEquals(URL, endpointInfo.url)
```
Comment (yift-r3) [inline libs/membership/membership-impl/src/test/kotlin/net/corda/membership/lib/impl/EndpointInfoImplTest.kt]: Can we change it to `assertThat`?

**Review by yift-r3 — DISMISSED** [review]
LGTM, one minor comment/question

**Review by yift-r3 — DISMISSED** [review]
LGTM

**Review by yift-r3 — DISMISSED** [review]

**Review by yift-r3 — APPROVED** [review]

**Review by alexey-kadyrov-r3 — APPROVED** [review]
LGTM

**Pull request merged**